File: src/main/java/com/hypherionmc/orion/plugin/paper/OrigamiExtension.java
package com.hypherionmc.orion.plugin.paper;

import lombok.Getter;
import org.gradle.api.Project;
import org.gradle.api.provider.ListProperty;
import org.gradle.api.provider.Property;

import java.util.ArrayList;

@Getter
public class OrigamiExtension {

    private final ListProperty<String> excludedPackages;
    private final ListProperty<String> excludedResources;
    private final Property<String> commonProject;

    private final Project project;

    public OrigamiExtension(Project project) {
        this.project = project;

        this.excludedPackages = project.getObjects().listProperty(String.class).convention(new ArrayList<>());
        this.excludedResources = project.getObjects().listProperty(String.class).convention(new ArrayList<>());
        this.commonProject = project.getObjects().property(String.class).convention("Common");
    }

}

File: src/main/kotlin/com/hypherionmc/orion/plugin/paper/OrigamiExtension.kt
/*
 * This file is part of orion, licensed under the MIT License (MIT).
 *
 * Copyright (c) 2025 HypherionSA and Contributors
 *
 */
package com.hypherionmc.orion.plugin.paper

import org.gradle.api.Project
import org.gradle.api.provider.ListProperty
import org.gradle.api.provider.Property

/**
 * @author HypherionSA
 *
 * Gradle Extension for the Origami Plugin, that handles which resources and classes should be
 * excluded when working with Modded code in Paper Plugins
 */
open class OrigamiExtension(project: Project) {

    val excludedPackages: ListProperty<String> = project.objects.listProperty(String::class.java).convention(emptyList())
    val excludedResources: ListProperty<String> = project.objects.listProperty(String::class.java).convention(emptyList())
    val commonProject: Property<String> = project.objects.property(String::class.java).convention("Common")

}